docs/rfc/active/data-permissions/README.md

```diff
@@ -0,0 +1,138 @@
+Datahub should provide functionality for viewing these permissions.
+
+## Motivation
+
+Within many sql systems, the data control language provides commands like GRANT, DENY and REVOKE to manage control to datasets.
+
+Being able to view grants is the first step toward supporting larger integrations with dataset permissions. [See Here](https://feature-requests.datahubproject.io/b/feedback/p/access-control-on-datasets-bi-assets)
+
+Of the datahub integration sources listed, the following have some form of data access control management:
+
+- Athena (via IAM)
+- BigQuery
+- Clickhouse
+- Druid
+- Elastic Search
+- Glue
+- Hive (via Apache Ranger)
+- Kafka
+- Looker
+- MariaDB
+- Metabase
+- MongoDB
+- MicrosoftSQLServer
+- MySQL
+- Oracle
+- PostgreSQL
+- Redash
+- Redshift
+- S3 (via IAM)
+- Sagemaker
+- Snowflake
+- Superset
+- Tableau
+- Trino
+
+## Requirements
+
+The permissions aspect must be:
+
+- Appliable to datasets, containers, dashboards
+- Support ingesting and viewing structured access control (GRANT {permission} ON {object} TO {user or role}
+- Support ingesting and viewing unstructured access control (IAM Policy Documents, Custom Policies from other datasets)
+-
+
+### Extensibility
+
+> Please also call out extensibility requirements. Is this proposal meant to be extended in the future? Are you adding
+> a new API or set of models that others can build on in later? Please list these concerns here as well.
+
+## Non-Requirements
+
+This RFC specifically calls for functionality to view the permissions. Actions taken on the permissions may be great areas for future enhancement but is not a focus here.
+
+## Detailed design
+
+There are no new Entities associated with this design.
+
+There is one aspect proposed:
+
+```
+record DataPermission {
+  structuredPermissions: array[StructuredDataPermission]
+  unstructuredPermissions: array[UnstructuredDataPermission]
+  relatedPermissions: array[Urn]
+}
+
+record StructuredDataPermission {
+   permission: string
+   object: Urn // if not specified, is implicitly the urn of the containing Entity
+   subject: union[corpUser, corpGroup, string] // grant to a user, group, or non-standard entity (e.g. "*", "all", "NONE")
+}
+
+record UnstructuredDataPermission {
+  language: string // maybe ENUM? (e.g. "IAM")
+  value: string
+}
+```
+
+This aspect should be attachable to
+
+- Datasets (e.g. SQL Grants on Tables)
+- Containers (e.g. Grants on Schemas/Databases)
+- Dashboards (e.g. Tableau permissions on dashboard access)
+- Charts (e.g. Tableau permissions on chart access)
+
+The `relatedPermissions` attribute allows us to expand Permissions into a full fledged Entity instead of just an aspect.
+
+The common case (SQL as an example) is primarily attached permissions (e.g. grants on a table or inline IAM policies).
+
+The extension allows for out of band policies (e.g. an external policy attached to multiple entities).
```

Comment:
Could you please elaborate how you differentiate between "attached" vs. "out of band"? In my understanding, this distinction can possibly be seen in the UI of some tools based on which "page" the policy is specified (e.g. the resource or the principal) but not sure if it is a fundamental conceptual distinction. In some cases e.g. in AWS services, we could enable access on both ends, the principal (IAM) and the resource (say S3 bucket policy), but semantically isn't the information the same?

(If all policies can, in a sense, be viewed as out-of-band i.e. existing/specified separately from the resource, without affecting the semantics, this would be additional motivation for modeling the DataPermission as an entity.)